src/aggregator/aggregator/aggregator.go

```diff
@@ -232,6 +239,46 @@ func (agg *aggregator) AddForwarded(
 	return nil
 }
 
+func (agg *aggregator) AddPassThrough(
+	metric aggregated.Metric,
+	metadata metadata.TimedMetadata,
+) error {
+	callStart := agg.nowFn()
+	agg.metrics.passThrough.Inc(1)
+	agg.RLock()
+	if agg.state != aggregatorOpen {
+		agg.metrics.addPassThrough.ReportError(errAggregatorAlreadyOpenOrClosed)
```

Comment:
unlock here too?

Reply:
+1, this needs to release the lock before returning